### ISSUE TITLE
SlevomatCodingStandard.Operators.NegationOperatorSpacing.InvalidSpaceAfterMinus false positive

### ISSUE THREAD

**Issue by morozov** [slevomat/coding-standard#1087]
```
$ composer show | grep -P 'coding-standard|php_codesniffer'

doctrine/coding-standard                       8.1.0  The Doctrine Coding Standard is a s...
slevomat/coding-standard                       6.4.0  Slevomat Coding Standard for PHP_Co...
squizlabs/php_codesniffer                      3.5.6  PHP_CodeSniffer tokenizes PHP, Java...
```
Consider the following code example:

```php
<?php

function test($x)
{
    return (int) $x->{'foo'} - (int) $x->{'bar'};
}
```
Try checking (false positive):
```
$ phpcs \
    -s \
    --standard=Doctrine \
    --sniffs=SlevomatCodingStandard.Operators.NegationOperatorSpacing,PSR12.Operators.OperatorSpacing \
    test.php

FILE: /home/morozov/Projects/phpcs-playground/test.php
------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------
 5 | ERROR | [x] Expected exactly 0 space after "-", 1 found.
   |       |     (SlevomatCodingStandard.Operators.NegationOperatorSpacing.InvalidSpaceAfterMinus)
------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------

Time: 63ms; Memory: 10MB
```
Try fo fix:
```
$ phpcbf \
    --standard=Doctrine \
    --sniffs=SlevomatCodingStandard.Operators.NegationOperatorSpacing,PSR12.Operators.OperatorSpacing \
    test.php

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
/home/morozov/Projects/phpcs-playground/test.php      FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 0 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 107ms; Memory: 10MB
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/4a0e3e8530cb08d7a8779fcfa55444794350d381
